### PR TITLE
Provide MessageFlushingStrategy (fixes #27)

### LINKED ISSUE #27: Call flush once per tick rather than after every send_message
Currently turbulence's channel `flush()` is called within each call to `NetworkResource::send_message` or `broadcast_message`. This will immediately send a packet containing the message.

If I understand correctly, one of the features of turbulence messages is that, given they must be less than the size of a packet, it coalesces them such that multiple messages are sent per packet.

By flushing each send, we are only ever sending 1 message per packet.

In the case of bevy, surely it would be better to `flush()` each channel once at the end of a tick, so that all the small messages that were sent by systems during that tick can be coalesced into fewer packets.

Perhaps exposing a flush_all_messages.system() that the user can run in an appropriate stage?

Thanks for a great library btw :)

### PR DESCRIPTION
**Pull request by RJ**
I kept the default behaviour the same as before, ie `MessageFlushingStrategy::OnEverySend`.

I think a better default would be `OncePerTick` but I don't see how to easily flush all channels in turbulence, I think we need https://github.com/kyren/turbulence/pull/4

I modifed the channels example to show usage.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -60,6 +61,7 @@ impl Plugin for NetworkingPlugin {
         app.insert_resource(NetworkResource::new(
             task_pool,
             self.link_conditioner.clone(),
+            self.message_flushing_strategy.clone(),
```
Comment (smokku) [inline src/lib.rs:64]: Is `clone` needed here?

**Comment by smokku** [comment]
> I modifed the channels example to show usage.

I'm not sure about this.
People tend to copy examples as a base for implementation, so the default example should be as simple as possible.
I'd rather see separate example - with an explanation why would one want to implement additional system to flush messages.

**Comment by smokku** [comment]
> I'd rather see separate example - with an explanation why would one want to implement additional system to flush messages.

Or we can wait for kyren/turbulence#4 which would allow us to hide it behind another flushing strategy option.

File: src/lib.rs (hunk 1 above)
Comment (RJ) [reply src/lib.rs:64]: Nope, i will make MessageFlushingStrategy derive Copy and remove the clone

**Comment by RJ** [comment]
I'll make a separate example to demonstrate. I might have to add some instrumentation to the transport to track how many packets are sent/recvd, so the coalescing is obvious.

File: src/lib.rs (hunk 1 above)
Comment (smokku) [reply src/lib.rs:64]: The semantics are bit different. Copy means that type is trivially copyable via bitblit, Clone means that making another entity (clone) might need some work (i.e. duplicating a file descriptor).

**Comment by RJ** [comment]
Ok, i added a `message_coalescing` demo and associated readme that shows some output, and reverted my changes to the channels demo

**Review by smokku — APPROVED** [review]
Love it. 😍 

I am perfectionist though, so if you would change these two little things?

File: examples/message_coalescing.rs [hunk 2]
```diff
@@ -0,0 +1,221 @@
+use bevy::{
+    app::{App, ScheduleRunnerSettings, AppExit, EventWriter, CoreStage},
+    core::Time,
+    ecs::prelude::*,
+    MinimalPlugins,
+};
+// use bevy_networking_turbulence::{NetworkEvent, NetworkResource, NetworkingPlugin, Packet};
```
[the remark this answers is not in the document]
Comment (RJ) [reply examples/message_coalescing.rs]: oops, will do

File: examples/message_coalescing_readme.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
+# Notes on message coalescing example
```
[the remark this answers is not in the document]
Comment (RJ) [reply examples/message_coalescing_readme.md]: yup

**Comment by RJ** [comment]
No problem, there we go.

**Comment by smokku** [comment]
TY

**Pull request merged**